### PR TITLE
Operator Overloading for Constant and Variable classes

### PR DESCRIPTION
**Pull request by PulkitMishra**

### CODE REVIEW

**Comment by PulkitMishra** [comment]
@8hantanu @sidkothiyal let me know if this approach is fine. Would want to work more in case this approach is satisfactory
Also I hope that Codacy not passing won't be a big issue

**Comment by PulkitMishra** [comment]
@8hantanu @sidkothiyal Please review.

**Review by 8hantanu — CHANGES_REQUESTED** [review]
Can you add test cases for the following?
- _Constant - Expression_
- _Expression - Constant_
- _Constant * Expression_
- _Constant / Expression_
- _Expression / Constant_
- _Variable + Expression_
- _Variable - Expression_
- _Expression - Variable_
- _Variable * Expression_
- _Variable / Expression_
- _Expression / Variable_

**Comment by PulkitMishra** [comment]
@8hantanu please look.

File: tests/test_functions.py [hunk 1]
```diff
@@ -19,6 +20,81 @@ def test_Constant():
...
+    summation = constant1 - variable1
+    assert summation.__str__() == "{({5}-5{x}^{3})}"
+
+    constant1 = Constant(5)
+    constant2 = Constant(5)
+    summation = constant1 * constant2
+    assert summation.__str__() == "{25}"
+
+    constant1 = Constant(5)
+    exp1 = constant1 + Variable(5, 'x', 3)
```
Comment (8hantanu) [inline tests/test_functions.py]: Can you define `exp1` explicitly as `Expression` as you did in line 120?
You can make similar changes to lines 55, 63, 72, 88, 94 or add separate cases where you define them explicitly.

Comment (PulkitMishra) [reply tests/test_functions.py]: @8hantanu done

**Review by 8hantanu — APPROVED** [review]

**Pull request merged**